src/workflow/ArcanistDiffWorkflow.php

```diff
@@ -2824,6 +2832,34 @@ private function pushChangesToStagingArea($id) {
     return $refs;
   }
 
+  /**
+   * Finds the git remote name for given git remote url.
+   *
+   * @return string Remote name if there is any, or empty string otherwise.
+   */
+   private function getRemoteName($remote_url) {
+     list($git_remote_stdout, $git_remote_stderr) = execx('git remote -v');
+     $git_remote_list = explode("\n", $git_remote_stdout);
+
+     foreach ($git_remote_list as $key => $line) {
+       if (strlen($line) > 0) {
+         // Every line has the following format:
+         // <remote_name>TAB<remote_url>SPACE([fetch|push|...])
```

Comment:
As you know the pattern, you can use `preg_match` here with named capture groups. This would save some lines. Though this is not necessary.